### PR TITLE
many: remove usage of LaidOutStructure in Observe and remove unused member

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**

### CODE REVIEW

**Review by MiguelPires — APPROVED** [review]
LGTM

File: boot/assets.go [hunk 1]
```diff
@@ -274,8 +274,8 @@ type TrustedAssetsInstallObserver struct {
 // measured as part of the secure boot or the bootloader configuration.
 //
 // Implements gadget.ContentObserver.
-func (o *TrustedAssetsInstallObserver) Observe(op gadget.ContentOperation, affectedStruct *gadget.LaidOutStructure, root, relativeTarget string, data *gadget.ContentChange) (gadget.ContentChangeAction, error) {
-	if affectedStruct.Role() != gadget.SystemBoot {
+func (o *TrustedAssetsInstallObserver) Observe(op gadget.ContentOperation, partRole string, root, relativeTarget string, data *gadget.ContentChange) (gadget.ContentChangeAction, error) {
```
Comment (pedronis) [inline boot/assets.go:277]: nitpick; you could leave out the string type and chain with the root ..., same in other similar places

File: gadget/layout.go [hunk 2]
```diff
@@ -63,8 +63,6 @@ type LaidOutVolume struct {
 	// LaidOutStructure is a list of structures within the volume, sorted
 	// by their start offsets
 	LaidOutStructure []LaidOutStructure
-	// RootDir is the root directory for volume data
-	RootDir string
```
Comment (pedronis) [inline gadget/layout.go:67]: so this wasn't used?

File: gadget/update.go [hunk 3]
```diff
@@ -105,7 +107,7 @@ type ContentObserver interface {
 	// ContentWrite operation, returning ChangeIgnore indicates that the
 	// change shall be ignored. ChangeAbort is expected to be returned along
 	// with a non-nil error.
-	Observe(op ContentOperation, sourceStruct *LaidOutStructure,
+	Observe(op ContentOperation, partRole string,
```
Comment (pedronis) [inline gadget/update.go:110]: same about chaining

**Review by pedronis — APPROVED** [review]
lgtm, some small comments

File: boot/assets.go (hunk 1 above)
Comment (alfonsosanchezbeato) [reply boot/assets.go:277]: Good catch, done

File: gadget/layout.go (hunk 2 above)
Comment (alfonsosanchezbeato) [reply gadget/layout.go:67]: It wasn't... must have been a leftover :man_shrugging:

**Pull request merged**